src/base/static/components/templates/map.tsx

```diff
@@ -486,28 +493,27 @@ class MapTemplate extends Component<Props, State> {
             mapViewport={this.state.mapViewport}
             onUpdateInitialMapViewport={this.props.onUpdateInitialMapViewport}
             onUpdateMapViewport={this.onUpdateMapViewport}
-            onUpdateMapDraggedOrZoomed={this.onUpdateMapDraggedOrZoomed}
-            onUpdateSpotlightMaskVisibility={
-              this.onUpdateSpotlightMaskVisibility
+            onUpdateMapDraggedOrZoomedByUser={
+              this.onUpdateMapDraggedOrZoomedByUser
             }
             onUpdateSourceLoadStatus={this.onUpdateSourceLoadStatus}
+            onUpdateMapTransitioning={this.onUpdateMapTransitioning}
+            isMapTransitioning={this.state.isMapTransitioning}
           />
           {this.props.isSpotlightMaskVisible && <SpotlightMask />}
         </div>
         {this.props.isContentPanelVisible && (
           <ContentPanel
             addPlaceButtonHeight={this.state.addPlaceButtonHeight}
-            isMapDraggedOrZoomed={this.state.isMapDraggedOrZoomed}
             currentLanguageCode={this.props.currentLanguageCode}
             defaultLanguageCode={this.props.defaultLanguageCode}
             mapContainerRef={this.mapContainerRef}
             mapViewport={this.state.mapViewport}
             onUpdateMapViewport={this.onUpdateMapViewport}
-            updateMapDraggedOrZoomed={isMapDraggedOrZoomed =>
-              this.setState({
-                isMapDraggedOrZoomed,
-              })
+            onUpdateMapDraggedOrZoomedByUser={
+              this.onUpdateMapDraggedOrZoomedByUser
             }
+            isMapTransitioning={this.state.isMapTransitioning}
```

Comment:
I was planning to move non-viewport map state (like the transition state) to the new map abstractions when they're ready.

Reply:
They should be ready now - what do you think about adding them to the `mapViewport` object?

Note that this would allow you to read their values from any connected component using the Map duck like so: https://github.com/mapseed/platform/blob/ac6d9c5e16293c7bb74fe6c18f8619ddd8efcc98/src/base/static/components/input-form/index.js#L696

But we are performing updates to the mapViewport through MainMap, not Redux, so you'd have to use the event emitter pattern, like we are doing here: https://github.com/mapseed/platform/blob/ac6d9c5e16293c7bb74fe6c18f8619ddd8efcc98/src/base/static/components/input-form/index.js#L86

Happy to clarify anything if needed.

Reply:
I considered using Redux instead of an event emitter to dispatch updates to the MapViewport state, but I thought the event emitter was simpler, since the source of truth is the mapViewport state in the MainMap component. Any changes to mapViewport state in MainMap are dispatched to the Redux store, via a throttled function.

If the event emitter pattern is difficult to manage, I may consider refactoring the updates to mapViewport state in MainMap to go through Redux. That would give us the advantage of managing the events through Redux. We can do this by storing an "incoming mapViewport" and an "outgoing mapViewport" in Redux.

Reply:
sounds good-- thanks for the explanation 👍 . I was curious how performance was managed in the new map state-- throttling makes sense. I haven't had time to look at it yet but I'll do so shortly...

Reply:
> what do you think about adding them to the mapViewport object?

I think we can't update transition and drag/zoom state in the same way viewport state is update in Redux (i.e. throttled), because we run the risk of throttling out key state changes (for example, the update when `isTransitioning` changes from `true` to `false`).

So I think we should just store non-viewport map state as separate pieces of information in the map duck. Non-viewport state updates much less frequently than viewport state, so I'm hoping there won't be any perf issues.

Reply:
>we run the risk of throttling out key state changes (for example, the update when isTransitioning changes from true to false).

I'm not sure I follow. It looks like `isMapTransitioning` is being stored as a component attribute: https://github.com/mapseed/platform/blob/373ffa459ce76fc3fca89910ad47d8b3fc7d166e/src/base/static/components/organisms/main-map.tsx#L173

What's the benefit of moving this into the map duck? It seems to only be used within the MainMap component.

FWIW I think it would be better to store this in the component state (outside of mapViewport), instead of a component attribute. That way, changes to `isMapTransitioning` would trigger a re-render.

Reply:
> What's the benefit of moving this into the map duck? It seems to only be used within the MainMap component.

This PR changes that requirement-- the `LngLatField` needs to know if the map is transitioning, since we want to avoid false positives in the drag map validation check.